client/src/pages/admin/CreateProduct.test.js

```diff
@@ -0,0 +1,689 @@
+import React from "react";
+import { render, fireEvent, waitFor, screen } from "@testing-library/react";
+import axios from "axios";
+import { BrowserRouter as Router } from "react-router-dom";
+import "@testing-library/jest-dom/extend-expect";
+import toast from "react-hot-toast";
+import CreateProduct from "./CreateProduct";
+import { expect } from "@jest/globals";
+import { act } from "react-dom/test-utils"; 
+
+jest.mock("axios");
+jest.mock("react-hot-toast");
+
+jest.mock("../../context/auth", () => ({
+  useAuth: jest.fn(() => [null, jest.fn()]),
+}));
+
+jest.mock("../../context/cart", () => ({
+  useCart: jest.fn(() => [null, jest.fn()]),
+}));
+
+jest.mock("../../context/search", () => ({
+  useSearch: jest.fn(() => [{ keyword: "" }, jest.fn()]),
+}));
+
+jest.mock("../../hooks/useCategory", () => jest.fn(() => []));
+
```

Comment:
```suggestion
jest.spyOn(console, "log").mockImplementation(() => {});
```

Perhaps we could mock the console log here to reduce cluttering